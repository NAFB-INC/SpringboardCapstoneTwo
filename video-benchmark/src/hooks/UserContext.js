import React from "react";

//storage place for app-wide data
const UserContext = React.createContext();

export default UserContext;
