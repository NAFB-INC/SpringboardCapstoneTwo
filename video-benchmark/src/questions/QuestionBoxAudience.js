import "./css/QuestionBox.css";
import React, { useEffect, useState } from "react";
import QuestionForm from "../forms/QuestionForm";

function QuestionBoxAudience({questions,setQuestions,fetchQuestions,presID}) {
    const [revealQBox,setRevealQBox] = useState(false);
    const [feedback,setFeedback] = useState("");

    useEffect(()=>{
        setTimeout(()=>{
            if(feedback){
                setFeedback("");
            }
        },5000)
    },[feedback,setFeedback]);

    return (
        <div className="QuestionBoxAudience">
            <div>{feedback}</div>
            {revealQBox?
                <div className="q-bounder">
                    <QuestionForm 
                        questions={questions} 
                        setQuestions={setQuestions} 
                        setRevealQBox={setRevealQBox} 
                        setFeedback={setFeedback} 
                        fetchQuestions={fetchQuestions} 
                        presID={presID}
                    />
                    <button id="nvm-q" className="present-choice-btn-sm" onClick={()=>{setRevealQBox(!revealQBox)}}>Never Mind</button>
                </div>
                :
                <div className="q-bounder">
                    <div>Have a question for the presenter? Ask away!</div>
                    <button id="have-q" className="present-choice-btn-2" onClick={()=>{setRevealQBox(!revealQBox)}}>I have a question!</button>
                </div>
            }
        </div>
    );
}

export default QuestionBoxAudience;