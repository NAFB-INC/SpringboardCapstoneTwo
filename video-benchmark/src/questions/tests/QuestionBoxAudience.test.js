import React from "react";
import { render } from '@testing-library/react';
import QuestionBoxAudience from '../QuestionBoxAudience';
import UserContext from "../../hooks/UserContext";

let questions = {};
const testCase1 = (
<div>
    <UserContext.Provider value={{ questions }}>
        <QuestionBoxAudience />
    </UserContext.Provider>
</div>);

it("renders without crashing", function() {
  render(testCase1);
});

it("matches snapshot", function() {
  const { asFragment } = render(testCase1);
  expect(asFragment()).toMatchSnapshot();
});