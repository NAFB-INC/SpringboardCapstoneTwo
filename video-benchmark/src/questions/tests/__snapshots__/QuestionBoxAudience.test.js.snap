// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="QuestionBoxAudience"
  >
    <div />
    <div
      class="q-bounder"
    >
      <div>
        Have a question for the presenter? Ask away!
      </div>
      <button
        class="present-choice-btn-2"
        id="have-q"
      >
        I have a question!
      </button>
    </div>
  </div>
</DocumentFragment>
`;
